src/boot/efi/meson.build

```diff
@@ -3,6 +3,10 @@
 efi_config_h_dir = meson.current_build_dir()
 efi_addon = ''
 
+if conf.get('EFI_MACHINE_TYPE_NAME', '') == ''
```

Comment:
this is set if efi_arch is set and bootloader is not explicitly set to false so it should work for CI too @medhefgo ?